runtime/caml/memory.h

```diff
@@ -211,33 +211,44 @@ int caml_page_table_initialize(mlsize_t bytesize);
 #define DEBUG_clear(result, wosize)
 #endif
 
-#define Alloc_small_with_profinfo(result, wosize, tag, profinfo) do { \
-  CAMLassert ((wosize) >= 1); \
-  CAMLassert ((tag_t) (tag) < 256); \
-  CAMLassert ((wosize) <= Max_young_wosize); \
-  caml_young_ptr -= Whsize_wosize (wosize); \
-  if (caml_young_ptr < caml_young_trigger){ \
-    caml_young_ptr += Whsize_wosize (wosize); \
-    CAML_INSTR_INT ("force_minor/alloc_small@", 1); \
-    Setup_for_gc; \
-    caml_gc_dispatch (); \
-    Restore_after_gc; \
-    caml_young_ptr -= Whsize_wosize (wosize); \
-  } \
-  Hd_hp (caml_young_ptr) = \
+extern void caml_alloc_small_dispatch (tag_t tag, intnat wosize, int track);
+#define Alloc_small_aux(result, wosize, tag, profinfo, track) do {     \
+  CAMLassert ((wosize) >= 1);                                          \
+  CAMLassert ((tag_t) (tag) < 256);                                    \
+  CAMLassert ((wosize) <= Max_young_wosize);                           \
+  caml_young_ptr -= Whsize_wosize (wosize);                            \
+  if (caml_young_ptr < caml_young_limit) {                             \
```

Comment:
Why did the previous test in the existing code use `caml_young_ptr < caml_young_trigger` instead of comparing to `caml_young_limit`? If I understand correctly, the change is safe because it will only dispatch more often, which is the right thing to do in the case where `caml_young_limit` is *not* `caml_young_trigger` (because it means that a signal is pending or a GC round has been requested).

Reply:
In bytecode mode, the two variables used to be equal.

In native code mode, you are almost right: this is correct because we would only dispatch more often. There is a subtlety, though, because `caml_alloc_small_dispatch` do not check for signals, nor for `caml_requested_major_slice`/`caml_requested_minor_gc`. I'll add that.